xoinvader/animation.py

```diff
@@ -0,0 +1,130 @@
+"""Animation.
+
+Animation is set of keyframes.
+Value of selected attribute changes in time.
+
+Keyframe:
+  (time, value)
+
+Objects have animation manager which manages animation graph and switching."""
+
+
+from operator import itemgetter
+
+from xoinvader.utils import Timer
+
+
+# TODO: Implement animation graph and etc
+class AnimationManager(object):
+    """Manage list of object animation."""
+
+    def __init__(self):
+        self._animations = {}
+        self._animation = None
+
+    @property
+    def animation(self):
+        """AnimationManager's current animation.
+
+        :getter: yes
+        :type getter: str
+        :setter: yes
+        :type setter: :class:`xoinvader.animation.Animation`
+        """
+
+        if self._animation:
+            return self._animation.name
+        else:
+            raise AttributeError("There is no available animation.")
+
+    @animation.setter
+    def animation(self, name):
+        if name in self._animations:
+            self._animation = self._animations[name]
+        else:
+            raise ValueError("No such animation: '{0}'.".format(name))
+
+    def add(self, name, *args, **kwargs):
+        """Add new animation, pass args to Animation class.
+
+        :param str name: animation name
+        """
+
+        animation = Animation(name, *args, **kwargs)
+        self._animations[name] = animation
+
+        if len(self._animations) == 1:
+            self._animation = animation
+
+    def update(self):
+        """Update manager's state."""
+
+        try:
+            self._animation.update()
+        except StopIteration:
+            raise  # We need to change animation
+
+
+class Animation(object):
+    """Animation unit.
+
+    :param str name: animation name
+    :param object bind: object to bind animation
+    :param str attr: attribute to change in frames
+    :param list keyframes: (float, object) tuples
+    :param bool loop: loop animation or not
+    """
+
+    def __init__(self, name, bind, attr, keyframes, loop=False):
+        self._name = name
+        self._obj = bind
+        self._attr = attr
+
+        if not keyframes:
+            raise ValueError("Animation keyframes must not be empty.")
+        self._keyframes = sorted(keyframes, key=itemgetter(0))
+
+        self._loop = loop
+
+        # Timer for tracking local time
+        self._timer = Timer(self._keyframes[-1][0], lambda: True)
```

Comment:
It's quite bad.
How we will stop the time? In paused state for example.
It's not the scope of the current issue however, but I think it would be nicer to implement somthing like is currently managing background advance.